mysqld_exporter.go

```diff
@@ -124,40 +125,66 @@ func (e *Exporter) scrape(scrapes chan<- []string) {
 	// fetch slave status
 	rows, err = db.Query("SHOW SLAVE STATUS")
 	if err != nil {
-		log.Println("error running show slave query on database: ", err)
+		log.Println("error running show slave query on database:", err)
 		e.error.Set(1)
 		e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
 		return
 	}
 	defer rows.Close()
 
+	var slaveCols []string
+
+	slaveCols, err = rows.Columns()
+	if err != nil {
+		log.Println("error retrieving column list:", err)
+		e.error.Set(1)
+		e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
+		return
+	}
+
+	var slaveData = make([]sql.RawBytes, len(slaveCols))
+
+	scanArgs := make([]interface{}, len(slaveCols))
```

Comment:
interface{} means any type (like Object in Java), so I think you should be able to use slaveData directly in the rows.Scan.